ls1intum/Artemis
# Feature: programming guided tour

### Checklist

### Description
This implementation offers the tutorial for the course `Tutorial Course` and exercise `Hello World` in Artemis.This implementation checks if the course and exercise that are available match with the given `courseShortName` and `exerciseShortName` in the guided tour and then automatically starts the guided tour if the tour has not been finished before.

### Steps for Testing
1. Log in to Artemis
2. Navigate to Course Overview
3. If the course `Tutorial Course` with the short name `artemistutorial` is available then the tour will be automatically displayed (if you see the tour for the first time, else start if from the account-menu), if not then nothing happens
4. Navigate through the tour to the course detail page
5. If the course `Hello World` with the short name `tutorial` is available then the tour will be automatically displayed, if not then nothing happens (you can only start the exercise when you rename the short name of the course to `artemis` - this is currently a bug which will be fixed in #866 )
6. Solve the given task in the local IDE or in the code editor and submit solution
7. Solve all tasks and then navigate to detail page again to see final tour step

### Screenshots
Course icon
![image](https://user-images.githubusercontent.com/33764166/65806223-d3662780-e188-11e9-8719-6ddbdea1d2de.png)

![start](https://user-images.githubusercontent.com/33764166/65839548-87dd8600-e30e-11e9-92ef-f984e3f04d0d.gif)
![end](https://user-images.githubusercontent.com/33764166/65839558-8ca23a00-e30e-11e9-84c9-0ef12c47d7b5.gif)
